src/featbin/paste-vectors.cc

```diff
@@ -0,0 +1,138 @@
+// featbin/paste-vectors.cc
+
+// Copyright 2012 Korbinian Riedhammer
+//           2013 Brno University of Technology (Author: Karel Vesely)
+//           2013 Johns Hopkins University (Author: Daniel Povey)
+//           2020 Ivan Medennikov (STC-innovation Ltd)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "matrix/kaldi-matrix.h"
+
+namespace kaldi {
+
+// returns true if successfully appended.
+bool AppendVectors(const std::vector<Vector<BaseFloat> > &in,
+                 std::string utt,
+                 Vector<BaseFloat> *out) {
+  // Check the lengths
+  int32 tot_dim = in[0].Dim();
+  for (int32 i = 1; i < in.size(); i++) {
+    int32 dim = in[i].Dim();
+    tot_dim += dim;
+  }
+  out->Resize(tot_dim);
+  int32 dim_offset = 0;
+  for (int32 i = 0; i < in.size(); i++) {
+    int32 this_dim = in[i].Dim();
+    out->Range(dim_offset, this_dim).CopyFromVec(
+        in[i].Range(0, this_dim));
+    dim_offset += this_dim;
+  }
+  return true;
+}
+
+
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace std;
+
+    const char *usage =
+        "Paste vector files \n"
```

Comment:
This is not specific: maybe 'Concatenate vectors' might be clearer?

Reply:
@danpovey, we have `concat-*` vs. `paste-*` semantics, and this one looks like it's `paste-*`-type? I.e., for
vec1
```
utt-1 [ 1 2 ]
utt-2 [ 3 4 5 ]
```
and vec2
```
utt-1 [ 11 12 ]
utt-2 [ 13 14 15 ]
```
paste (vec1 vec2) is
```
utt-1 [ 1 2 11 12 ]
utt-2 [ 3 4 5 13 14 15 ]
```
Did I grok what the tool was doing?

Reply:
From a  UNIX-command-line-manipulating text point of view you're right, but (a) these aren't text files necessarily, (b) someone may not make that mental connection.  These are vectors, "concatenate" is something that makes sense from a vector standpoint.

Reply:
@danpovey should I rename paste-vectors to concat-vectors, or you meant only usage string?

Reply:
@danpovey, I'm kinda torn on this, the naming would be inconsistent either. We already have binaries with a similar semantics called `paste-*`, and dissimilar called `concat-*`. Maybe `combine-vectors` would work?

Yes, of course I understand these are not usually text files.